xbmc/settings/SettingsComponent.cpp

```diff
@@ -47,11 +47,11 @@ void CSettingsComponent::Init(const CAppParamParser &params)
   if (m_state == State::DEINITED)
   {
     // only the InitDirectories* for the current platform should return true
-    bool inited = InitDirectoriesLinux(params.m_platformDirectories);
+    bool inited = InitDirectoriesLinux(params.GetPlatformDirectories());
     if (!inited)
-      inited = InitDirectoriesOSX(params.m_platformDirectories);
+      inited = InitDirectoriesOSX(params.GetPlatformDirectories());
     if (!inited)
-      inited = InitDirectoriesWin32(params.m_platformDirectories);
+      inited = InitDirectoriesWin32(params.GetPlatformDirectories());
```

Comment:
This is more just a comment. I noticed this block when i was looking at clang-analyze reports a couple weeks back, and it throws an unused variable on the last instance. As its settings, its early in the startup, but itd be nice to pull these out into platform specific over-rides or something one day

Reply:
hmm that's interesting. Yea it would be nice to split more of the platform code out.